modules/hcp-hc-objects.adoc

```diff
@@ -0,0 +1,295 @@
+spec: {}
+status: {}
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  creationTimestamp: null
+  name: <hosted_cluster_namespace> <2>
+spec: {}
+status: {}
+----
++
+<1> Replace `<hosted_cluster_name>` with your hosted cluster.
+<2> Replace `<hosted_cluster_namespace>` with the name of your hosted cluster namespace.
+
+. Create a YAML file with the following information about the config maps and secrets to include in the `HostedCluster` deployment:
++
+[source,yaml]
+----
+---
+apiVersion: v1
+data:
+  ca-bundle.crt: |
+    -----BEGIN CERTIFICATE-----
+    -----END CERTIFICATE-----
+kind: ConfigMap
+metadata:
+  name: user-ca-bundle
+  namespace: <hosted_cluster_namespace> <1>
+---
+apiVersion: v1
+data:
+  .dockerconfigjson: xxxxxxxxx
+kind: Secret
+metadata:
+  creationTimestamp: null
+  name: <hosted_cluster_name>-pull-secret <2>
+  namespace: <hosted_cluster_namespace> <1>
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: sshkey-cluster-<hosted_cluster_name> <2>
+  namespace: <hosted_cluster_namespace> <1>
+stringData:
+  id_rsa.pub: ssh-rsa xxxxxxxxx
+---
+apiVersion: v1
+data:
+  key: nTPtVBEt03owkrKhIdmSW8jrWRxU57KO/fnZa8oaG0Y=
+kind: Secret
+metadata:
+  creationTimestamp: null
+  name: <hosted_cluster_name>-etcd-encryption-key <2>
+  namespace: <hosted_cluster_namespace> <1>
+type: Opaque
+----
++
+<1> Replace `<hosted_cluster_namespace>` with the name of your hosted cluster namespace.
+<2> Replace `<hosted_cluster_name>` with your hosted cluster.
+
+. Create a YAML file that contains the RBAC roles so that Assisted Service agents can be in the same `HostedControlPlane` namespace as the hosted control plane and still be managed by the cluster API:
++
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  creationTimestamp: null
+  name: capi-provider-role
+  namespace: <hosted_cluster_namespace>-<hosted_cluster_name> <1> <2>
+rules:
+- apiGroups:
+  - agent-install.openshift.io
+  resources:
+  - agents
+  verbs:
+  - '*'
+----
++
+<1> Replace `<hosted_cluster_namespace>` with the name of your hosted cluster namespace.
+<2> Replace `<hosted_cluster_name>` with your hosted cluster.
+
+. Create a YAML file with information about the `HostedCluster` object, replacing values as necessary:
++
+[source,yaml]
+----
+apiVersion: hypershift.openshift.io/v1beta1
+kind: HostedCluster
+metadata:
+  name: <hosted_cluster_name> <1>
+  namespace: <hosted_cluster_namespace> <2>
+spec:
+  additionalTrustBundle:
+    name: "user-ca-bundle"
+  olmCatalogPlacement: guest
+  imageContentSources: <3>
+  - source: quay.io/openshift-release-dev/ocp-v4.0-art-dev
+    mirrors:
+    - registry.<dns.base.domain.name>:5000/openshift/release <4>
+  - source: quay.io/openshift-release-dev/ocp-release
+    mirrors:
+    - registry.<dns.base.domain.name>:5000/openshift/release-images <4>
+  - mirrors:
+  ...
+  ...
+  autoscaling: {}
+  controllerAvailabilityPolicy: SingleReplica
```

Comment:
@davidvossel so we need this for the Kubevirt provider?

Reply:
We (KubeVirt) support SingleReplica and HighlyAvailable.